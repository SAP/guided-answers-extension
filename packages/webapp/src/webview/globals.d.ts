interface Window {
    vscode: {
        postMessage: (message: string | JSON) => void;
    };
}
