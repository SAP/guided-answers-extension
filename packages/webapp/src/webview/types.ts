import type {
    GuidedAnswerNode,
    GuidedAnswerTree,
    GuidedAnswerTreeSearchResult
} from '@sap/guided-answers-extension-types';

export interface AppState {
    loading: boolean;
    query: string;
    guidedAnswerTreeSearchResult: GuidedAnswerTreeSearchResult;
    activeGuidedAnswerNode: GuidedAnswerNode[];
    activeGuidedAnswer?: GuidedAnswerTree;
    betaFeatures: boolean;
    searchResultCount: number;
    guideFeedback: null | boolean;
    selectedProductFilters: string[];
    selectedComponentFilters: string[];
    pageSize: number;
    feedbackStatus: boolean;
    feedbackResponse: boolean;
}
