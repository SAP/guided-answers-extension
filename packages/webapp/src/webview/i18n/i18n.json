{
    "GUIDED_ANSWER_HEADER": "Guide",
    "SEARCH_GUIDED_ANSWERS": "Search Guided Answers",
    "ALL_ANSWERS": "Home",
    "STEP_BACK": "Step back",
    "RESTART": "Restart",
    "GUIDED_ANSWERS": "Guided Answers",
    "NO_ANSWERS_FOUND": "No answers found",
    "PLEASE_MODIFY_SEARCH": "Please modfiy search",
    "THIS_SOLVED_MY_ISSUE": "This solved my issue",
    "THIS_DID_NOT_SOLVED_MY_ISSUE": "This did not solve my issue",
    "THANK_YOU_TEXT": "We are glad to hear that your issue has been resolved and we hope you enjoy using the Guided Answers application.",
    "THANKS": "Thanks!",
    "HOME": "Home",
    "SEND": "Send",
    "CLOSE": "Close",
    "ISSUE_IS_NOT_RESOLVED": "Issue is not resolved",
    "WE_ARE_SORRY_TO_HEAR_THAT_YOUR_ISSUE_IS_NOT_YET_RESOLVED": "We are sorry to hear that your issue is not yet resolved.",
    "PLEASE_TELL_US_IF_THIS_ANSWER_WAS_HELPFUL": "Please tell us if this answer was helpful",
    "THERE_ARE_SEVERAL_OPTIONS_FOR_GETTING_FURTHER_ASSISTANCE": "There are several options for getting further assistance:",
    "SEARCH_FOR_ANOTHER_GUIDED_ANSWER": "Search for another Guided Answer",
    "START_AN_EXPERT_CHAT": "Start an Expert Chat",
    "SCHUEDULE_AN_EXPERT": "Schedule an Expert",
    "OPEN_AN_INCIDENT": "Open an Incident",
    "ASK_THE_SAP_COMMUNITY": "Ask the SAP Community",
    "FEEDBACK": "Feedback",
    "FEEDBACK_DIALOG_TITLE": "Is this content helpful?",
    "FEEDBACK_DIALOG_SUBTEXT": "If you have suggestions on how to improve this content we would love to hear them!",
    "FEEDBACK_DIALOG_DISCLAIMER": "Your feedback is anonymous, we do not collect any personal data.",
    "FEEDBACK_DIALOG_SUGGESTION": "Your suggestion",
    "FEEDBACK_SENT_DIALOG_TITLE": "Message sent",
    "FEEDBACK_SENT_DIALOG_SUBTEXT": "Thanks for the feedback!"
}
