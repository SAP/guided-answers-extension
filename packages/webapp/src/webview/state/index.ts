export { actions, store } from './store';
export { getInitialState } from './reducers';
