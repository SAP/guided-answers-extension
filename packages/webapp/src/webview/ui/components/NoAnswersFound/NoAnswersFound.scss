* {
    font-family: var(--vscode-font-family);
}
.error-screen {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 100px;

    &__objects {
        display: flex;
        flex-direction: column;
        align-items: center;
    }

    h1 {
        font-style: normal;
        font-weight: 700;
        font-size: 24px;
        line-height: 29px;
        margin-top: 0px;
        margin-bottom: 0px;
    }

    h2 {
        font-style: normal;
        font-weight: 200;
        font-size: 13px;
        line-height: 18px;
        margin: 20px 0px 20px 0px;
    }

    &__logo {
        margin-top: 20px;
    }
}
