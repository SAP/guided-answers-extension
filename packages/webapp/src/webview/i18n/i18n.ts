import { use } from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as i18nEn from './i18n.json';

export { i18n } from 'i18next';

export function initI18n(): void {
    use(initReactI18next)
        .init({
            resources: {
                en: {
                    translation: i18nEn
                }
            },
            lng: 'en',
            fallbackLng: 'en',
            interpolation: {
                escapeValue: false
            }
        })
        .catch((error) => console.error(error));
}
