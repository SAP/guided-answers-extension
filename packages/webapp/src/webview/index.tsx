import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { App } from './ui/components/App';
import { actions, store } from './state';
import { initI18n } from './i18n';

/**
 * Initialization of i18n, icons
 */
initI18n();

/**
 * Render app
 */
render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root') as HTMLElement
);

/**
 * Fire action which is captured in hosting app (ide-extension) to signal that we are ready to receive messages
 */
actions.webviewReady();

/**Once webview is ready we focus to search field */
document.getElementById('search-field')?.focus();
