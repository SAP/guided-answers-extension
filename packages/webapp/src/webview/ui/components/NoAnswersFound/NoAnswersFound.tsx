import React from 'react';
import './NoAnswersFound.scss';
import ErrorIcon from './no-answers-found.svg';
import i18next from 'i18next';

/**
 *@returns - react element to show an error page when there are no answers found
 */
export function NoAnswersFound() {
    return (
        <div className="error-screen">
            <div className="error-screen__objects">
                <h1 className="error-screen__object__title">{i18next.t('NO_ANSWERS_FOUND')}</h1>
                <h2 className="error-screen__object__subtitle">{i18next.t('PLEASE_MODIFY_SEARCH')}</h2>
                <span className="error-screen__object__logo">
                    <ErrorIcon />
                </span>
            </div>
        </div>
    );
}
