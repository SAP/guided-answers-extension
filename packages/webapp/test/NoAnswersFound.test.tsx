import React from 'react';
import { shallow } from 'enzyme';
import { NoAnswersFound } from '../src/webview/ui/components/NoAnswersFound';
import i18next from 'i18next';
import { initI18n } from '../src/webview/i18n';

describe('<NoAnswersFound />', () => {
    let wrapper: any;
    initI18n();
    beforeEach(() => {
        wrapper = shallow(<NoAnswersFound />);
    });

    afterEach(() => {
        jest.clearAllMocks();
        wrapper.unmount();
    });

    it('Should render a NoAnswersFound component', () => {
        expect(wrapper.find('.error-screen').length).toBe(1);
        expect(wrapper.find('.error-screen__object__title').text()).toBe(i18next.t('NO_ANSWERS_FOUND'));
        expect(wrapper.find('.error-screen__object__subtitle').text()).toBe(i18next.t('PLEASE_MODIFY_SEARCH'));
    });
});
