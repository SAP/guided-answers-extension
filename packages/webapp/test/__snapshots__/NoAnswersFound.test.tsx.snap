// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoAnswersFound /> Should render a NoAnswersFound component 1`] = `
<div>
  <div
    class="error-screen"
  >
    <div
      class="error-screen__objects"
    >
      <h1
        class="error-screen__object__title"
      >
        No answers found
      </h1>
      <h2
        class="error-screen__object__subtitle"
      >
        Please modfiy search
      </h2>
      <span
        class="error-screen__object__logo"
      >
        <svg />
      </span>
    </div>
  </div>
</div>
`;
